core/store/src/archive/cold_storage.rs

```diff
@@ -399,23 +423,45 @@ pub fn test_get_store_initial_writes(column: DBCol) -> u64 {
 /// So, for every KeyType we need to capture all the keys that are related to that block.
 /// For BlockHash it is just one key -- block hash of that height.
 /// But for TransactionHash, for example, it is all of the tx hashes in that block.
+///
+/// Although `tracked_shards` should be sufficient to determine which shards matter,
+/// we process all shards as a precaution to ensure robustness and avoid
+/// missing data in case `tracked_shards` are incomplete due to bugs or future
+/// changes. This redundancy is acceptable because only the State column is
+/// performance- and size-sensitive, and it is handled separately in `copy_state_from_store`.
 fn get_keys_from_store(
     store: &Store,
     shard_layout: &ShardLayout,
+    tracked_shards: &Vec<ShardUId>,
```

Comment:
Just to make sure - you only use tracked_shards for sanity checks here, is that right?